### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by jbohanon**
Small fixes caused by bad merging (status redeclaration) and version incompatibilites with 1.28 (localInfo)

### CODE REVIEW

File: source/extensions/filters/http/ext_proc/config.cc
Comment (ben-taussig-solo) [inline source/extensions/filters/http/ext_proc/config.cc:22]: I'm assuming this function was renamed upstream -- is that correct?

File: source/extensions/filters/http/ext_proc/ext_proc.cc
Comment (ben-taussig-solo) [inline source/extensions/filters/http/ext_proc/ext_proc.cc:755]: Did the function signature of `createActivation` change in upstream?

Comment (jbohanon) [reply source/extensions/filters/http/ext_proc/ext_proc.cc:755]: yes, in [this PR](https://github.com/envoyproxy/envoy/pull/31319/files#diff-e2f19282a4d26ecf20fdfb6b5e386ecd7a57545f03b77f32ff2f9efde1d5190e), so we will need the localInfo once we bump to 1.29 ([link](https://github.com/envoyproxy/envoy/blob/v1.29.1/source/extensions/filters/common/expr/evaluator.h#L69))

File: source/extensions/filters/http/ext_proc/config.cc
Comment (jbohanon) [reply source/extensions/filters/http/ext_proc/config.cc:22]: yeah the `getServerFactoryContext` is the old function for this, which was changed [here](https://github.com/envoyproxy/envoy/pull/31150/files#diff-6eb911fbc2179b0e242b66ffb4d97d814b014d348c3f0c70d80d8db039e6e779), which should get picked up in 1.29

File: source/extensions/filters/http/ext_proc/ext_proc.cc
Comment (ben-taussig-solo) [reply source/extensions/filters/http/ext_proc/ext_proc.cc:755]: Ah, I see. Are we currently tracking changes that we know we'll have to make when we bump to 1.29? If not, do you think it makes sense to leave an inline comment including a link to that PR/description of what changes will be necessary? Or will that just make things harder when we have to get the 1.29 bump out?

Comment (jbohanon) [reply source/extensions/filters/http/ext_proc/ext_proc.cc:755]: I think it would be simpler to make the 1.29 changes from a 1.30-dev cherry-pick (which is what I did here), and since that release was cut pmuch right before these merged, these divergences won't be an issue.

Basically, if we are keel-hauling these changes forward through versions, we should track them, but if we cherry-pick from merged upstream it won't be an issue

**Review by nfuden — APPROVED** [review]
Seems to build and work as expected

**Review by ben-taussig-solo — APPROVED** [review]
I don't think we need a second approval but here's one anyway

**Pull request merged**
